clients/python-wrapper/lakefs/branch.py

```diff
@@ -1,12 +1,112 @@
+        """
+        Create a new branch in lakeFS from this object
+        :param source_reference_id: The reference to create the branch from
+        :param exist_ok: If False will throw an exception if a branch by this name already exists. Otherwise,
+         return the existing branch without creating a new one
+        :return: The lakeFS SDK object representing the branch
+        :raises
+            NotAuthorizedException if user is not authorized to perform this operation
+            ServerException for any other errors
+        """
+
+        def handle_conflict(e: LakeFSException):
+            if isinstance(e, ConflictException) and exist_ok:
+                return None
+            return e
+
+        branch_creation = lakefs_sdk.BranchCreation(name=self._id, source=source_reference_id)
+        with api_exception_handler(handle_conflict):
+            self._client.sdk_client.branches_api.create_branch(self._repo_id, branch_creation)
+        return self
+
+    def head(self) -> Reference:
+        """
+        Get the commit reference this branch is pointing to
+        :return: The commit reference this branch is pointing to
+        :raises
+            NotFoundException if branch by this id does not exist
+            NotAuthorizedException if user is not authorized to perform this operation
+            ServerException for any other errors
+        """
+        with api_exception_handler():
+            branch = self._client.sdk_client.branches_api.get_branch(self._repo_id, self._id)
+        return Reference(self._repo_id, branch.commit_id, self._client)
+
+    def commit(self, message: str, metadata: dict = None) -> Reference:
+        """
+        Commit changes on the current branch
+        :param message: Commit message
+        :param metadata: Metadata to attach to the commit
+        :return: The new reference after the commit
+        :raises
+            NotFoundException if branch by this id does not exist
+            NotAuthorizedException if user is not authorized to perform this operation
+            ServerException for any other errors
+        """
+
+        commits_creation = lakefs_sdk.CommitCreation(message=message, metadata=metadata)
+        with api_exception_handler():
+            c = self._client.sdk_client.commits_api.commit(self._repo_id, self._id, commits_creation)
+        return Reference(self._repo_id, c.id, self._client)
+
+    def delete(self) -> None:
+        """
+        Delete branch from lakeFS server
+        :raises
+            NotFoundException if branch by this id does not exist
+            NotAuthorizedException if user is not authorized to perform this operation
+            ServerException for any other errors
+        """
+        with api_exception_handler():
+            return self._client.sdk_client.branches_api.delete_branch(self._repo_id, self._id)
+
+    def revert(self, reference_id: str, parent_number=0) -> None:
```

Comment:
Good catch, since parent_number is a required param - is default=0 a main usecase? Otherwise consider leaving without default

Reply:
Good thing you asked 😄 
It is the default value, but it's a bit confusing. In lakectl the flag is optional
- if it is called without a value, it calls the API with value 0
- if it's called with value 0 it throws an error
This is the same in git 
I am implementing the same